crowbar_framework/app/assets/javascripts/jquery/nodeList.js

```diff
@@ -317,6 +321,40 @@
             break;
         }
       }
+
+      if (constraints.platform !== undefined && constraints.platform && !cluster) {
+        var platforms = $.map(constraints.platform, function(c_version, c_platform) {
+          return self.equalOrMatch(platform, c_platform) && self.equalOrMatch(platform_version, c_version);
+        });
+        var is_any_true = platforms.some(function (element, index, array) {
+          return element;
+        });
+        if (!is_any_true) {
+          return self.errorMessage(
+            'barclamp.node_selector.platform'.localize().format(
+              alias,
+              role
+            )
+          );
+        }
+      }
+
+      if (constraints.exclude_platform !== undefined && constraints.exclude_platform && !cluster) {
```

Comment:
No need for the cluster check as it doesn't have a platform exclude constraint attribute.